src/lib/stepper/stepper.spec.ts

```diff
@@ -0,0 +1,767 @@
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {FormControl, FormGroup, ReactiveFormsModule, Validators} from '@angular/forms';
+import {MdStepperNext, MdStepperPrevious} from './stepper-button';
+import {dispatchKeyboardEvent} from '@angular/cdk/testing';
+import {ENTER, LEFT_ARROW, RIGHT_ARROW, SPACE} from '@angular/cdk/keycodes';
+import {MdStepper} from './stepper';
+
+const EMAIL_REGEX = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+describe('MdHorizontalStepper', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MdStepperModule, NoopAnimationsModule, ReactiveFormsModule],
+      declarations: [
+        SimpleMdHorizontalStepperApp,
+        LinearMdHorizontalStepperApp
+      ]
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  describe('basic horizontal stepper', () => {
+    let fixture: ComponentFixture<SimpleMdHorizontalStepperApp>;
+    let stepperComponent: MdHorizontalStepper;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(SimpleMdHorizontalStepperApp);
+      fixture.detectChanges();
+
+      stepperComponent = fixture.debugElement
+          .query(By.css('md-horizontal-stepper')).componentInstance;
+    });
+
+    it('should default to the first step', () => {
+      expect(stepperComponent.selectedIndex).toBe(0);
+    });
+
+    it('should change selected index on header click', () => {
+      let stepHeaders = fixture.debugElement.queryAll(By.css('.mat-horizontal-stepper-header'));
+      assertSelectionChangeOnHeaderClick(stepperComponent, fixture, stepHeaders);
+    });
+
+    it('should set the "tablist" role on stepper', () => {
+      let stepperEl = fixture.debugElement.query(By.css('md-horizontal-stepper')).nativeElement;
+      expect(stepperEl.getAttribute('role')).toBe('tablist');
+    });
+
+    it('should set aria-expanded of content correctly', () => {
+      let stepContents = fixture.debugElement.queryAll(By.css(`.mat-horizontal-stepper-content`));
+      assertCorrectAriaExpandedAttribute(stepperComponent, fixture, stepContents);
+    });
+
+    it('should display the correct label', () => {
+      assertCorrectStepLabel(stepperComponent, fixture);
+    });
+
+    it('should go to next available step when the next button is clicked', () => {
+      assertNextStepperButtonClick(stepperComponent, fixture);
+    });
+
+    it('should go to previous available step when the previous button is clicked', () => {
+      assertPreviousStepperButtonClick(stepperComponent, fixture);
+    });
+
+    it('should set the correct step position for animation', () => {
+      assertCorrectStepPosition(stepperComponent, fixture);
+    });
+
+    it('should support keyboard events to move and select focus', () => {
+      let stepHeaders = fixture.debugElement.queryAll(By.css('.mat-horizontal-stepper-header'));
+      assertCorrectKeyboardInteraction(stepperComponent, fixture, stepHeaders);
```

Comment:
Not a fan of having the assertions be in separate functions, because it makes failures harder to track down, although in this case the assertions seem pretty complicated. Thoughts @jelbourn?